### ISSUE TITLE
Send MouseButtonReleased events when the mouse is outside the window on Windows

### ISSUE THREAD

**Issue by Wizzard033** [SFML/SFML#455]
I was attempting to request a feature from TGUI, the SFML GUI, here in this thread http://forum.tgui.eu/index.php?topic=123

However, Texus, developer of TGUI, says that the feature I requested already exists on Linux and only does not work on Windows because of an inconsistency in SFML.
The problem is that MouseButtonReleased events are not sent to properly close out a MouseButtonPressed event when the mouse is dragged outside the window.
Texus says in that thread that in Linux MouseButtonReleased events are sent correctly in this case.

Put simply, if one is to click and hold the left mouse button inside a SFML window on Windows, then move the mouse outside of the SFML window and stop holding the left mouse button down, only a MouseButtonPressed event happened.
This is supposedly inconsistent with Linux and is definitely inconsistent with what I think should happen.
